python/pyspark/ml/base.py

```diff
@@ -47,6 +86,28 @@ def _fit(self, dataset):
         """
         raise NotImplementedError()
 
+    @since("2.3.0")
+    def fitMultiple(self, dataset, params):
```

Comment:
So in Scala Spark we use the `fit` function rather than separate functions. Also the `params` name is different than the Scala one. Any reason for the difference?

Reply:
Check out the discussion on the JIRA and the linked design doc.  Basically, we need the same argument types but different return types from what the current fit() method provides.

Reply:
We couldn't use `fit` because it's going to have the same signature as the existing `fit` method but return a different type, (Iterator[(Int, Model)] instead of Seq[Model]). I was trying to be consistent with Estimator.fit which uses the name `params` which is different than the name of the same argument in Scala :/. Happy to change it.

Reply:
That's a good point that we could rename "params" to be clearer.  How about "paramMaps"?

Reply:
I made this change.